### PR TITLE
Added Testing for HTTP Method Overriding

### LINKED ISSUE #446: HTTP Method overriding testing
I am proposing to add a new testing in the Test HTTP Methods section called 'Testing for HTTP Method Overriding'. This will encompass the usage of alternative HTTP headers to override the actual HTTP method in the request.

Would you like to be assigned to this issue?
Check the box if you will submit a PR to add the proposed content. Please read CONTRIBUTING.md.
- [x] Assign me, please!

### PR DESCRIPTION
**Pull request by irgoncalves**
This PR closes #446.

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Add 'Testing for HTTP Method Overriding'

### CODE REVIEW

**Comment by kingthorin** [comment]
> document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:209 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]

File: document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md [hunk 1]
```diff
@@ -172,6 +172,40 @@ If the tester thinks that the system is vulnerable to this issue, they should is
 
 With some luck, using the above three commands - modified to suit the application under test and testing requirements - a new user would be created, a password assigned, and made an administrator, all using blind request submission.
 
+### Testing for HTTP Method Overriding
+
+Some web frameworks provide a way to override the actual HTTP method in the request  by emulating the missing HTTP verbs passing some custom header in the requests. The main purpose of this is to circumvent some middleware (e.g. proxy, firewalls) limitation where methods allowed usually do not encompass verbs such as PUT or DELETE for instance. The following alternative headers could be used to do verb tunneling:
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:177]:
```suggestion
Some web frameworks provide a way to override the actual HTTP method in the request by emulating the missing HTTP verbs passing some custom header in the requests. The main purpose of this is to circumvent some middleware (e.g. proxy, firewall) limitation where methods allowed usually do not encompass verbs such as `PUT` or `DELETE`. The following alternative headers could be used to do such verb tunneling:
```

File: document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md [hunk 2]
```diff
@@ -172,6 +172,40 @@ If the tester thinks that the system is vulnerable to this issue, they should is
...
+
+- X-HTTP-Method
+- X-HTTP-Method-Override
+- X-Method-Override
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:181]:
```suggestion
- `X-HTTP-Method`
- `X-HTTP-Method-Override`
- `X-Method-Override`
```

File: document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md [hunk 3]
```diff
@@ -172,6 +172,40 @@ If the tester thinks that the system is vulnerable to this issue, they should is
...
+
+In order to test this, in the scenarios where restrictive verbs such as PUT or DELETE returns a “405 Method not allowed” replay the same request with the addition of the alternative headers for HTTP method overriding and observe how the web server will respond. The webserver should respond with a different status code (e.g. 200) in case method overriding is supported:
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:183]:
```suggestion
In order to test this, in the scenarios where restrictive verbs such as PUT or DELETE return a “405 Method not allowed”, replay the same request with the addition of the alternative headers for HTTP method overriding, and observe how the system will respond. The application should respond with a different status code (e.g. 200) in cases where method overriding is supported:
```

File: document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md [hunk 4]
```diff
@@ -172,6 +172,40 @@ If the tester thinks that the system is vulnerable to this issue, they should is
...
+
+$ nc www.example.com 80
+DELETE /resource.html HTTP/1.1
+Host: www.example.com
+X-HTTP-Method: DELETE
+
+HTTP/1.1 200 OK
+Date: Sat, 04 Apr 2020 19:26:01 GMT
+Server: Apache
+
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:206]:
```suggestion
```

**Comment by irgoncalves** [comment]
Thanks @kingthorin. I'll do it.

**Comment by kingthorin** [comment]
Those 4 items still need to be addressed.

**Comment by irgoncalves** [comment]
@kingthorin , all changes request look good and accepted.

**Comment by irgoncalves** [comment]
@ThunderSon Thanks. Yes, I'll add it.

**Comment by irgoncalves** [comment]
@ThunderSon @kingthorin , references updated.

**Pull request merged**